src/components/securemessage/SecureMessage1L.java

```diff
@@ -0,0 +1,172 @@
+package components.securemessage;
+
+import components.map.Map;
+import components.map.Map1L;
+
+/**
+ * Kernel Implementation of the SecureMessage component.
+ *
+ * @convention $this.encryptedMap has non-null keys and values, and each
+ *             MessageKey has non-empty message and key strings.
+ *
+ * @correspondence this = mapping from each (message, key) pair in
+ *                 $this.encryptedMap to its associated encrypted message
+ */
+public class SecureMessage1L extends SecureMessageSecondary {
+
+    /**
+     * Helper class representing a (message, key) pair.
+     */
+    private static class MessageKey {
```

Comment:
If this is immutable, I might recommend a Java record. It comes with the common methods for free.